File: 2302. Count Subarrays With Score Less Than K.cpp
class Solution {
public:
    long long countSubarrays(vector<int>& nums, long long k) 
    {
        long long ans = 0, sum = 0;
        int i = 0, j = 0, n = nums.size();
        while(j < n)
        {
            sum += nums[j];
            while(sum * (j - i + 1) >= k) 
                sum -= nums[i++];
            ans += j - i + 1;
            j++;
        }
        return ans;
    }
};

File: 2302. Count Subarrays With Score Less Than K.py
class Solution:
    def countSubarrays(self, nums: List[int], k: int) -> int:
        i, j, sum = 0, 0, 0
        ans, n = 0, len(nums)
        while j < n:
            sum += nums[j]
            while (j - i + 1) * sum >= k:
                sum -= nums[i]
                i += 1
            ans += j - i + 1
            j += 1
        return ans